### DISCUSSION TITLE
Question: Location of MSVC runtime dll's when `include_msvcr` is True

### DISCUSSION THREAD

**Discussion by FreeHarry** [marcelotduarte/cx_Freeze#1065] [Q&A] [answered]
Just for my understanding: How does cx_freeze find the location of the MSVC runtime dll's specified in module `winmsvcr`? Which logic is used here?

I can include these files with the option `include_msvcr`. But when I check from where the files are copied on my environment, I figured out they come from the location `C:\Program Files\PowerShell\7`. It seems the finder looks anywhere in the `%ProgramFiles%` directory and add these file where they was founded first. Is that correct? What is if there are other locations which different states of these files. Which location is preferred?

Would it be possible to add more information about this topic to the documentation of cx_freeze? From my opinion it would be important to know/understand from which location they are copied.

**Comment by marcelotduarte** [comment, marked as answer]
> Which logic is used here?

The logic of windows for searching dll. There are a list of MSVC dlls, if the you use include_msvcr=False, these dlls are ignored. If true, the dlls are searched in the %PATH% as windows does.

**Comment by FreeHarry** [comment]
Ok, Thanks for the info. Is this the same also for `vcruntime140.dll` and `vcruntime140_1.dll`?

**Comment by marcelotduarte** [comment]
Yes, it is.
The list of dll is here: https://github.com/marcelotduarte/cx_Freeze/blob/main/cx_Freeze/winmsvcr.py
But for these 2 ddls has a caveat that it should be along python38.dll and the executable.

To be honest, in python 3.6-3.7 the search is on the PATH. For python 3.8-3.9 has another approach, except in anaconda and msys2, which the search continues in the PATH.

Are you using anaconda or python from python.org? What version?

**Comment by marcelotduarte** [comment]
Some docs:
https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-setdlldirectoryw
https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-adddlldirectory
